src/main/java/com/ibm/devops/connect/CloudSocketComponent.java

```diff
@@ -155,10 +191,42 @@ public void handleDelivery(String consumerTag, Envelope envelope,
                         boolean connected = CloudPublisher.testConnection(syncId, syncToken, url);
                     } else {
                         String message = new String(body, "UTF-8");
-                        System.out.println(" [x] Received '" + message + "'");
-
-                        CloudWorkListener2 cloudWorkListener = new CloudWorkListener2();
-                        cloudWorkListener.call("startJob", message);
+                        String payload = null;
+                        String syncToken = getSyncToken();
+                        try {
+                            payload = decrypt(syncToken, message.toString());
+                        } catch (Exception e) {
+                            //TODO handle decryption error
+                            System.out.println("Unable to decrypt");
+                        }
+                        //TODO Don't make this an array in the silly way that I have.  I just want this to work
+                        JSONArray incomingJobs = JSONArray.fromObject("[" + payload + "]");
+                        JSONObject incomingJob = incomingJobs.getJSONObject(0);
+                        String workId = incomingJob.getString("id");
+                        String jobName = incomingJob.getString("fullName");
+                        try {
+                            StandardUsernamePasswordCredentials credentials = Jenkins.getInstance().getDescriptorByType(DevOpsGlobalConfiguration.class).getCredentialsObj();          
+                            String plainCredentials = credentials.getUsername() + ":" + credentials.getPassword().getPlainText();
+                            String encodedString = getEncodedString(plainCredentials);
+                            String authorizationHeader = "Basic " + encodedString;
+                            String rootUrl = Jenkins.getInstance().getRootUrl();
+                            HttpResponse<String> response = Unirest.get(rootUrl+"job/"+jobName.replaceAll(" ", "%20")+"/lastBuild/consoleText")
+                                .header("Authorization", authorizationHeader)
+                                .asString();
+                            String lastBuildConsole = response.getBody().toString();
+                            boolean isFound = lastBuildConsole.contains(workId); // true
```

Comment:
So is this logic making a rest call to jenkins to get the console output for the last build? Seems like we shouldn't need to make a rest call for that, shouldn't that be possible just using `hudson` class objects? Also, just checking for the `id` being printed out in the last build seems pretty "hacky" and not a sure-fire way of knowing its a duplicate.

Reply:
In the case of duplicate jobs getting triggered, the duplicate jobs are getting triggered at a very small time difference. We are not storing workId for different jobs. The only place from where I was able to retrieve the workId was console logs. I'm not sure if we can get console logs of a build while it's running using huson class objects. For making sure that we are searching for the right thing instead of searching just for workId we can try to search for the complete string mentioned in the image attached (_Started due to a request from UrbanCode Velocity. Work Id: NEA82yDgTMvtqzxmu_14_) 
<img width="662" alt="Screenshot 2021-03-08 at 11 01 35 PM" src="https://user-images.githubusercontent.com/79622980/110359447-823bb200-8063-11eb-91ee-a2756b8c09b9.png">

Reply:
Also, if a job has not yet built (no lastBuild), in that case, the Hunson class object might not work, but rest call will return some response of 'page not found'. This response won't have any effect on rest of the process.

Reply:
https://stackoverflow.com/questions/37386581/jenkins-console-output-log-location-in-filesystem
<img width="747" alt="Screenshot 2021-03-15 at 7 44 33 PM" src="https://user-images.githubusercontent.com/79622980/111167282-f5df4100-85c6-11eb-994e-8cfeda896fac.png">

Reply:
Hudson model takes console log from LogFile, which gets written once the build ends. So we can't get real-time logs for a running build.

Reply:
Another way of getting real-time logs is using Jenkins-cli.jar

Reply:
@aboe026, To me, it seems that using rest API is the most simple and the safest way.